scipy/stats/tests/test_hypotests.py

```diff
@@ -858,6 +858,70 @@ def test_contingency_table_return(self):
         res = stats.somersd(x, y)
         assert_equal(res.table, np.eye(10))
 
+    def test_somersd_alternative(self):
+        # Test alternative parameter, asymptotic method (due to tie)
+
+        # Based on scipy.stats.test_stats.TestCorrSpearman2::test_alternative
+        x1 = [1, 2, 3, 4, 5]
+        x2 = [5, 6, 7, 8, 7]
+
+        # strong positive correlation
+        expected = stats.somersd(x1, x2, alternative="two-sided")
+        assert expected.statistic > 0
+
+        # rank correlation > 0 -> large "less" p-value
+        res = stats.somersd(x1, x2, alternative="less")
+        assert_equal(res.statistic, expected.statistic)
+        assert_allclose(res.pvalue, 1 - (expected.pvalue / 2))
+
+        # rank correlation > 0 -> small "greater" p-value
+        res = stats.somersd(x1, x2, alternative="greater")
+        assert_equal(res.statistic, expected.statistic)
+        assert_allclose(res.pvalue, expected.pvalue / 2)
+
+        # reverse the direction of rank correlation
+        x2.reverse()
+
+        # strong negative correlation
+        expected = stats.somersd(x1, x2, alternative="two-sided")
+        assert expected.statistic < 0
+
+        # rank correlation < 0 -> large "greater" p-value
+        res = stats.somersd(x1, x2, alternative="greater")
+        assert_equal(res.statistic, expected.statistic)
+        assert_allclose(res.pvalue, 1 - (expected.pvalue / 2))
+
+        # rank correlation < 0 -> small "less" p-value
+        res = stats.somersd(x1, x2, alternative="less")
+        assert_equal(res.statistic, expected.statistic)
+        assert_allclose(res.pvalue, expected.pvalue / 2)
+
+        with pytest.raises(ValueError, match="alternative must be 'less'..."):
+            stats.somersd(x1, x2, alternative="ekki-ekki")
+
+    def test_somersd_perfect_correlation(self):
```

Comment:
How about adding perfectly negative correlation test just like you have done in `test_somersd_alternative`:

```py
x1 = list(range(10))
x2 = x1.copy()
x2.reverse()

# perfectly negative correlation -> small "two-sided" p-value (0)
res = stats.somersd(x1, x2, alternative="two-sided")
assert res.statistic == -1
assert res.pvalue == 0

# rank correlation < 0 -> small "less" p-value (0)
res = stats.somersd(x1, x2, alternative="less")
assert res.statistic == -1
assert res.pvalue == 0

# rank correlation < 0 -> large "greater" p-value (1)
res = stats.somersd(x1, x2, alternative="greater")
assert res.statistic == -1
assert res.pvalue == 1
```